examples/react/src/App.tsx

```diff
@@ -1,24 +1,106 @@
+/* eslint-disable jsx-a11y/media-has-caption */
 /* eslint-disable react/button-has-type */
 /* eslint-disable react/jsx-props-no-spreading */
 /* eslint-disable react/react-in-jsx-scope */
-import { useState } from 'react'
+import { useEffect, useState, useRef } from 'react'
 import {
   Dropzone,
   FilesGrid,
   FilesList,
   ProviderIcon,
   UploadButton,
   UppyContextProvider,
+  useWebcam,
   useDropzone,
   useFileInput,
 } from '@uppy/react'
 import Uppy from '@uppy/core'
 import Tus from '@uppy/tus'
+import UppyWebcam from '@uppy/webcam'
 
 import './app.css'
 import '@uppy/react/dist/styles.css'
 
-function CustomDropzone() {
+interface WebcamProps {
+  isOpen: boolean
+  close: () => void
+}
+
+function Webcam({ isOpen, close }: WebcamProps) {
```

Comment:
if the consumers of the useWebcam hook , wanting to display errors (e.g modal) 
they'd probably do it this way I suppose ? 

```
const {state} = useWebcam 

  const renderError = () => {
    if (state.status === 'error') {
      let errorMessage = 'An unknown camera error occurred.'
      if (state.cameraError?.message) {
              errorMessage = `Camera error: ${state.cameraError.message}`
      }
      return (
        <div className="some-style-class">
          <p>{errorMessage}</p>
        </div>
      )
    }
    return null
  }
  ```

Reply:
Something like that can be done, yes